### DISCUSSION TITLE
Average 50% brightness drop?

### DISCUSSION THREAD

**Discussion by DynV** [ImageMagick/ImageMagick#6436] [Help]
### ImageMagick version

7.0.10-29

### Operating system, version and so on

Windows 10 Pro 22H2

### Description

I have a directory with images that are definitely brighter than I need. I'd like on average close to 50% of their brightness to be reduced; brighter ones reduced more and less bright reduced less. I'd like for the originals to remain the same and the result be in another directory.

Perhaps this could be done in 2 passes. The 1st determining their brightness (perhaps in a flat-file DB). Then some calculations could be done on my part to figure how much I reduce to which. Finally the 2nd pass does the brightness reduction.

I'd appreciate help or tips toward that goal

**Comment by snibgo** [comment]
What type of images are these? Photos? Can you show a sample?

How do you want to reduce brightness? Perhaps by multiplying pixel values by a number `N`, where `0 <= N <= 1`.

How do you determine `N` for each image? Perhaps you want to make the mean pixel value of the image 0.5, on a scale of 0 to 1. If so, you can calculate N by:
```
N = 0.5 / old_mean
```

The `magick` command could then be:
```
magick in.png -evaluate Multiply %[fx:0.5/mean] out.png
```

**Comment by DynV** [comment]
They are background images, mostly nature & gaming, but others themes which
I like the combination of color & texture; for example this is my latest
download batch:  https://imgur.com/KzHnJom Reducing brightness similar to
how I'd do it on GIMP: menu > Colors > Brightness-Contrast... > move slider
so # is -. To determine the degree, I guess I'd find 1-2 samples of the
average brightness I like and set my desktop so the samples are on 1 side
of the screen and GIMP the other side, then move the slider until I think
it matches the samples brightness. Is there a command so it calculates the
brightness of all the backgrounds in a directory, perhaps in a flat-file DB
(ie TSV)? The following URLs are of the same example background image
(multiple for archival, in case they break):

   -
   https://www.wallpaperflare.com/green-abstract-art-lines-stripes-backgrounds-pattern-illustration-wallpaper-twoe
   - https://wallpapersafari.com/w/ynTKNx
   -
   https://wallpapercrafter.com/717389-windows-windows-vista-home-vista-premium-backgrounds.html
   - https://www.pxfuel.com/en/desktop-wallpaper-xivwg
